pkgs/applications/window-managers/tinywl/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, wlroots, pkg-config
+, libxkbcommon, pixman, udev, wayland, wayland-protocols
+}:
+
+stdenv.mkDerivation {
+  pname = "tinywl";
+  inherit (wlroots) version src;
+
+  sourceRoot = "source/tinywl";
+
+  nativeBuildInputs = [ pkg-config ];
```

Comment:
Ideally also `wayland-scanner` (e.g. for cross-compilation, I haven't tested that though).